js/scripts/zone_loaders/village_zone_loaders.js

```diff
@@ -296,34 +307,59 @@ const loadVillageMain = () => {
-        GAME.addEntity(new BlackSmith(Vector.blockToWorldSpace(blockPosBlackSmith), new Conversation(getAllConversationArrays().village.blacksmith.merchant)), 0);
-        GAME.addEntity(new Mayor(Vector.blockToWorldSpace(blockPosMayor), new Conversation(getAllConversationArrays().village.mayor.hopefulGreeting)), 0);
-        // TODO add these in after you get back from the field.
-        // GAME.addEntity(new MamaChad(Vector.blockToWorldSpace(blockPosMama)));
-        // GAME.addEntity(new Wizard(Vector.blockToWorldSpace(blockPosWizard)));
-        GAME.addEntity(idleMama);
-
-        if (LAST_ZONE === null) {
+        /*
+        The above content was all static. Below, there are conditional spawns/settings based on story progression.
+        Namely, we need to script the village attack when the tutorial (snake/bunny hunt) is complete.
+        */
+
+        if (!STORY.tutorialComplete) {
+            // NPCs
+            const blockPosPapa = new Vector(33, chadOnGround);
+            const blockPosBlackSmith = new Vector(17, chadOnGround);
+            const blockPosMayor = new Vector(50, chadOnGround);
+            const blockPosIdleMama = new Vector(37, chadOnGround);
+
+            const idleMama = new MamaChad(Vector.blockToWorldSpace(blockPosIdleMama), false, new Conversation(getAllConversationArrays().village.mamaChad.goodMorning));
+            idleMama.action = "idle";
+
+            GAME.addEntity(new PapaChad(Vector.blockToWorldSpace(blockPosPapa), new Conversation(getAllConversationArrays().village.papaChad.huntingInvite)), 0);
+            GAME.addEntity(new BlackSmith(Vector.blockToWorldSpace(blockPosBlackSmith), new Conversation(getAllConversationArrays().village.blacksmith.merchant)), 0);
+            GAME.addEntity(new Mayor(Vector.blockToWorldSpace(blockPosMayor), new Conversation(getAllConversationArrays().village.mayor.hopefulGreeting)), 0);
+            GAME.addEntity(idleMama);
+
+            let weather = "warm";
+            WeatherSystem.setWeather(weather, 5, "day");
+        } else {
+            const blockPosTrappedMama = new Vector(65, chadOnGround + 1);
+            const blockPosWizard = new Vector(63, chadOnGround);
+            GAME.addEntity(new MamaChad(Vector.blockToWorldSpace(blockPosTrappedMama)));
+            GAME.addEntity(new Wizard(Vector.blockToWorldSpace(blockPosWizard)));
+            if (STORY.tutorialComplete && !STORY.villageAttackEnded) {
+                for (let blockx = 10; blockx < 60; blockx += 5) {
+                    console.log('adding a slime');
+                    GAME.addEntity(new Slime(Vector.blockToWorldSpace(new Vector(blockx, chadOnGround)), Slime.EVIL));
+                }
+            }
+            let weather = "rain";
+            WeatherSystem.setWeather(weather, 2, "day");
+        }
+        let surfaceSnow = false;
```

Comment:
I realize I likely made the interpreter default to snow being true as a fun little indication that something wasn't right but we should change that probably so we can clean up code like this. You shouldn't need to specify that second parameter and it's likely on me. Perhaps I'll add that to my list of updates this week.